src/python/serenity_types/pricing/derivatives/rates/yield_curve.py

```diff
@@ -147,6 +147,12 @@ class YieldCurveDefinition(BaseModel):
     The specific interpolator type used to bootstrap this curve.
     """
 
+    rate_source_type: RateSourceType
+    """
+    The type of input being provided for this CurvePoint, e.g. if it's from a 3M future,
+    this would be FUTURE_PX, while if it's from traditional rates fixings, it would be FIXINGS.
+    """
+
```

Comment:
I think need to add back rate_source_type: RateSourceType.
Code is using this and although it is in raw CurvePoint, no way to tie back to interpolated

Reply:
Can you explain? The YieldCurveVersion has both CurvePoints and the interpolated representation. What's missing?